include/ada/implementation.h

```diff
@@ -17,9 +17,15 @@ namespace ada {
    * @param base_url the optional string input to use as a base url.
    * @param encoding encoding (default to UTF-8)
    */
-  ada_warn_unused ada::url parse(std::string input,
+  ada_warn_unused ada::url parse(std::string& input,
+                                 /* Passing std::optional<ada::url> by copy likely forces a copy of the ada::url instance. */
                                  std::optional<ada::url> base_url = std::nullopt,
                                  ada::encoding_type encoding = UTF8) noexcept;
+  ada_warn_unused ada::url parse(std::string_view input,
+                                 /* Passing std::optional<ada::url> by copy likely forces a copy of the ada::url instance. */
+                                 std::optional<ada::url> base_url = std::nullopt,
```

Comment:
We can get rid of `std::optional` by overloading the function, and providing an alternative with just `(input, encoding)` parameters